### PR TITLE
Add type definitions for react-loadable

### PR DESCRIPTION
**Pull request by Jessidhia**
Mostly based off the Flow type annotations in the original code or readme snippets, but with some additional generics tricks for more accurate typing.

JSDoc comments copied off the README.

### CODE REVIEW

File: types/react-loadable/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
+// Type definitions for react-loadable 3.3
+// Project: https://github.com/thejameskyle/react-loadable#readme
+// Definitions by: Diogo Franco <https://github.com/Kovensky>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+import * as React from 'react';
+
+// tslint:disable-next-line strict-export-declare-modifiers
+type LoadedComponent<Props> = React.ComponentClass<Props> | React.SFC<Props>;
```
Comment (Kovensky) [inline types/react-loadable/index.d.ts]: This is just a shorthand alias for internal use, and not something I want to export; but the strict-export-declare-modifiers was demanding I added an `export` here.

If this is NG, then I'll have to change all instances of `LoadedComponent<Props>` into `(React.ComponentClass<Props> | React.SFC<Props>)` which can lead to some very long lines in the interfaces.

File: types/react-loadable/react-loadable-tests.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+interface ComponentProps {
+  text: string;
+}
+
+const Component: React.SFC<ComponentProps> = ({ text }) => <div>{text}</div>;
+
+const Loadable100 = Loadable({
+  // a module shape with 'export = Component' / 'module.exports = Component'
+  loader: () => Promise.resolve(Component),
```
Comment (Kovensky) [inline types/react-loadable/react-loadable-tests.tsx]: The `Component` and the various shapes can perhaps be split into separate test-specific files once dynamic import support is in typescript, for writing this test how it's intended to be used -- `() => import('./Component')`.

File: types/react-loadable/index.d.ts (hunk 1 above)
Comment (andy-ms) [inline types/react-loadable/index.d.ts]: It's exported whether you write the keyword there or not. Although, I don't see any problem with having a helpful type alias exposed.

Comment (Kovensky) [inline types/react-loadable/index.d.ts]: Hm, it is indeed.

I'll just export this one; but is there any way to make an unexported local alias?

This alias also looks helpful enough to maybe belong on react.d.ts.

Comment (andy-ms) [inline types/react-loadable/index.d.ts]: Sorry, I don't think there's a way to hide a part of a module declaration.

**Comment by andy-ms** [comment]
Thanks!

**Pull request merged**